### PR TITLE
Add UI section

### PR DESCRIPTION
**Pull request by ghys**
Add several pages related to UI.
Move sitemaps, HABPanel to it.
Rename "User Interfaces & Ecosystem" to "Ecosystem".
Rename "Beginner Tutorial" to "Getting Started".
Put the installation guide above the Concepts section.

Signed-off-by: Yannick Schaus <github@schaus.net>

### CODE REVIEW

**Review by Confectrician — COMMENTED** [review]
As a first feedback:

image

I have looked through the `ruby`, `js` and `image` files and found no issues so far.
Sidebar looks really tidied up in the preview.

I am now diving into the content.

File: ui/layout-pages.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+This is how the layout will adapt depending on the width of the screen:
+
+![Responsive Layout](./images/responsive2.gif)
+
+## How to Build a Layout Page
+
+### Defining the Layout
+
+When you create a new Layout Page, you'll notice the 2 buttons *"Add Block"* and *"Add Masonry"*.
```
Comment (Confectrician) [inline ui/layout-pages.md:100]: I would go for `Add Block` and `Add Masonry` (with single backticks) here.

That's how it is displayed in the preview:
image

File: ui/index.md [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+|[Map](./map-pages.html)|main UI|main UI|JSON DB|
+|[Floorplan](./floorplan-pages.html)|main UI|main UI|JSON DB|
+|[Chart](./chart-pages.html)|main UI|main UI|JSON DB|
+|[Tabbed](./tabbed-pages.html)|main UI|main UI|JSON DB|
+
+## Designing Pages
+
+You can see the list of your Pages in the administration area of the main UI under _Settings > Pages_, and from there, create new ones, as well as modify or remove existing ones.
+
+::: tip
```
Comment (Confectrician) [inline ui/index.md:78]:
```suggestion
::: tip Attention
```

**Review by Confectrician — COMMENTED** [review]
First two pages reviewed.

File: ui/index.md [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+
+You will likely notice, the more you use the user interface, that when you use it at all, you always come back to the same controls: that's because you (hopefully) made comprehensive automation rules to make sure all you need to get to the state that you want is triggering those rules - and this can be done with a mere button. Those are the critical controls that you should focus on when designing your home page.
+
+Of course, you will also probably need to dive into details from time to time, but everything doesn't have to fit on this overview page, so you can also create more of them that will be displayed upon clicking on a button.
+
+### The Sidebar
+
+Another entry point to start navigating your Pages is to put some of them on the main UI's Sidebar: they will appear on the left side, or in the left drawer on smaller screens, just below the openHAB logo.
+
+To put a page on the Sidebar, open the "Sidebar & Visibility" option in a Page's general settings, and enable the "Show on Sidebar" toggle.
```
[the remark this answers is not in the document]
Comment (ghys) [reply ui/index.md]: Is there a formal guidance for these? IMO the backticks (monospace font) should be for code or technical terms only, like parameter identifiers or item names. Labels on the other hand should remain in the normal font, perhaps in italics or bold. I've tried replacing most occurrences from italics to bold, maybe it emphasizes better.

File: ui/layout-pages.md (hunk 1 above)
Comment (ghys) [reply ui/layout-pages.md]: Same remark as above. I think we should distinguish technical text and labels.

File: ui/index.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (Confectrician) [reply ui/index.md]: Generally i would agree an the code thing, but specially for ui purposes i like the "button like" look when it's rendered in monospace to help users recognize such terms. (Pls keep in mind that not everyone reading this docs is speaking "code" like a dev does)

But no, there is nor formal guidance, just some subjective influence from my side.

File: ui/building-pages.md [hunk 4]
```diff
@@ -0,0 +1,344 @@
...
+
+**UI components** are the basic building blocks for many UIs in openHAB 3.
+The Main UI Pages and Personal Widgets are notable UI components, but Sitemaps that were created in the UI and HABPanel dashboards are as well.
+
+These structures make up hierarchies that notably define the pages in their entirety are relatively simple:
+
+Each component has:
+
+- A **type**;
+- Most of the time, a set of **configuration properties**;
```
Comment (Confectrician) [inline ui/building-pages.md:15]: I would remove those semicolons after each list element, tough i am not sure about the correct grammar.
Just feels a bit wrong to me.

File: ui/building-pages.md [hunk 5]
```diff
@@ -0,0 +1,344 @@
...
+
+- `items` is a dynamic key/value dictionary allowing you to retrieve the state of items; the result of `items.Item1` will be an object like `{ state: '23', displayState: '23 °C' }` (`displayState` may be omitted). You can therefore use `items.Item1.state` to use the current state of Item1 in your expression, if it changes, it will be reevaluated;
+- `props` is a dictionary of the key/values of self-defined props for the current personal widget, or page (pages, like any root UI components, may indeed have props). It is indispensable to use props in expressions when developing a personal widget;
+- `vars` is a dictionary of variables (see below) that are available in the component's context;
+- `loop` is a dictionary containing iteration information when you're repeating components from a source collection, it is defined only when in the context of a `oh-repeater` component;
+- the JavaScript `Math` object (so your can use _Math.floor(...), Math.round(...) and the like;
```
Comment (Confectrician) [inline ui/building-pages.md:149]:
```suggestion
- the JavaScript `Math` object (so you can use _Math.floor(...), Math.round(...) and the like)
```

File: ui/building-pages.md [hunk 6]
```diff
@@ -0,0 +1,344 @@
...
+- [A Complete Guide to Grid](https://css-tricks.com/snippets/css/complete-guide-grid/)
+- [Grid Tutorial on W3Schools](https://www.w3schools.com/css/css_grid.asp)
+
+## Personal Widgets
+
+You can extend the library of widgets you have at your disposal by creating personal ones, either by yourself, or copy-pasting from examples by the community; then you can reuse them on pages, multiple times if need be, simply configuring their props to your needs.
+To add a new personal widget, as an admin, go to **Developer Tools > Widgets**, then use the '+' button to create a new one.
+
+The view features a code (YAML) editor and a live preview, you can change the orientation with the button in the center of the bottom toolbar.
+Don't forget to change the `uid` right away because you won't be able to alter it afterwards.
```
Comment (Confectrician) [inline ui/building-pages.md:330]: This should go in a tip or warning box

**Review by Confectrician — COMMENTED** [review]
Here are some small leftover comments.

Everything fine for me afterwards.
Feel free to use bold text instead of monospace for the unresolved review comments.
I think it doesn't make sense to mix that up and it is used also for technical parts wheere it makes sense.

File: ui/building-pages.md (hunk 4 above)
Comment (ghys) [reply ui/building-pages.md:15]: I've always done that as a habit, just checked and:

> Semicolons were traditionally used in lists to separate each bullet point, but although this is still correct, they are not used as often today. Legal writing is probably the main exception.
([source](https://www.onlinegrammar.com.au/punctuation-in-lists/)).

So I guess I'm old school, or a wannabe lawyer :)
But if they aren't used elsewhere I have no problem to remove them.

**Comment by ghys** [comment]
> Everything fine for me afterwards.

@Confectrician if there's nothing pressing anymore I'll merge this so I can make the changes on the website now and also integrate the component docs: https://github.com/openhab/openhab-docs/tree/final/_addons_uis/org.openhab.ui/doc/components.

**Pull request merged**